src/sentry/coreapi.py

```diff
@@ -335,10 +343,10 @@ def _process_data_timestamp(self, data, current_datetime=None):
             current_datetime = datetime.now()
 
         if value > current_datetime + timedelta(minutes=1):
-            raise InvalidTimestamp('Invalid value for timestamp (in future): %r' % value)
+            raise FutureInvalidTimestamp('%s' % (value - current_datetime))
 
         if value < current_datetime - timedelta(days=30):
-            raise InvalidTimestamp('Invalid value for timestamp (too old): %r' % value)
+            raise PastInvalidTimestamp('%s' % (value - current_datetime))
```

Comment:
You might be aware of this, but the string representation of a negative datetime that includes both days and a time component can be difficult to interpret unless you're familiar with it already:

```python
>>> now = datetime.datetime.now()
>>> past = now - datetime.timedelta(days=31, hours=6)
>>> str(past - now)
'-32 days, 18:00:00'
```

It might make more sense to just provide the delta as `(value - current_datetime).total_seconds()` here instead.